Mathlib/RingTheory/NoetherNormalization.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2025 Sihan Su. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca, Sihan Su, Wan Lin, Xiaoyang Su
+-/
+import Mathlib.Algebra.MvPolynomial.Monad
+import Mathlib.Data.List.Indexes
+import Mathlib.RingTheory.IntegralClosure.IsIntegralClosure.Basic
+/-!
+# Noether Normalizaton
+This file contains a proof of Noether normalization lemma by Nagata.
+
+## Main Results
+Let `A` be a finitely generated algebra over a field `k`.
+Then there exists natural number `r` and an injective homomorphism
+from `k[X_1, X_2, ..., X_r]` to `A` such that `A` is integral over `k[X_1, X_2, ..., X_r]`.
+
+## Reference
+* <https://stacks.math.columbia.edu/tag/00OW>
+
+-/
+open Polynomial MvPolynomial Ideal BigOperators Nat RingHom List
+
+variable {k : Type*} [Field k] {n : ℕ} (f : MvPolynomial (Fin (n + 1)) k)
+variable (v w : Fin (n + 1) →₀ ℕ)
+
+namespace NoetherNormalization
+
+/-- Suppose `f` is a nonzero polynomial of `n+1` variables : `X_0,...,X_n`.
+`up` is a number which is big enough. -/
+local notation3 "up" => 2 + f.totalDegree
+
+/-- `r` maps i to `up ^ i`-/
+local notation3 "r" => fun (i : Fin (n + 1)) ↦ up ^ i.1
+
+/-- We construct an algebra map `T1` which maps `X_i` into `X_i + X_0 ^ r_i` when `i ≠ 0`,
+and maps `f` into some monic polynomial (regarded as a polynomial of `X_0`).-/
```

Comment:
```suggestion
/-- We construct an algebra map `T1 f c` which maps `X_i` into `X_i + c • X_0 ^ r_i` when `i ≠ 0`
and `X_0` to `X_0`.-/
```
Here you say "It maps `f` into some monic polynomial (regarded as a polynomial of `X_0`).". Do you prove this later in some sense?

Reply:
How does `T1 f c` depend on `f`?

Reply:
Because `r` depends on `f`.